### PR TITLE
[Merged by Bors] - feat(CategoryTheory): construction of sections of functors by transfinite induction

### PR DESCRIPTION
**Pull request by joelriou**
Given a functor `F : Jᵒᵖ ⥤ Type v` where `J` is a well-ordered type, we introduce a structure `F.WellOrderInductionData` which allows to show that the map `F.sections → F.obj (op ⊥)` is surjective.

The data and properties in `F.WellOrderInductionData` consist of a section to the maps `F.obj (op (Order.succ j)) → F.obj (op j)` when `j` is not maximal, and, when `j` is limit, a section to the canonical map from `F.obj (op j)` to the type of compatible families of elements in `F.obj (op i)` for `i < j`.

In other words, from `val₀ : F.obj (op ⊥)`, a term `d : F.WellOrderInductionData` allows the construction, by transfinite induction, of a section of `F` which restricts to `val₀`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge
